pkg/alert/commands.go

```diff
@@ -318,3 +316,187 @@ func NewDeleteAlertSettingConditionCommand() *cobra.Command {
+		TraverseChildren: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			if err := bite.CheckRequiredFlags(cmd, bite.FlagPair{"name": channel.Name, "connectionName": channel.ConnectionName, "templateName": channel.TemplateName, "properties": channel.Properties}); err != nil {
+				return err
+			}
+
+			if propertiesRaw != "" {
+				if err := bite.TryReadFile(propertiesRaw, &channel.Properties); err != nil {
+					// from flag as json.
+					if err = json.Unmarshal([]byte(propertiesRaw), &channel.Properties); err != nil {
+						return fmt.Errorf("Unable to unmarshal the properties: [%v]", err)
+					}
+				}
+			}
+
+			if err := config.Client.CreateAlertChannel(channel.Name, channel.ConnectionName, channel.TemplateName, channel.Properties); err != nil {
+				golog.Errorf("Failed to create alert channel [%s]. [%s]", channel.Name, err.Error())
+				return err
+			}
+
+			return bite.PrintInfo(cmd, "Alert channel [%s] created", channel.Name)
+		},
+	}
+
+	cmd.Flags().StringVar(&channel.Name, "name", "", "Alert channel name")
+	cmd.Flags().StringVar(&channel.ConnectionName, "connectionName", "", "Alert channel connection name")
+	cmd.Flags().StringVar(&channel.TemplateName, "templateName", "", "Alert channel template name")
+	cmd.Flags().StringVar(&propertiesRaw, "properties", "", `Alert channel properties .e.g. "[{\"key\":\"username\",\"value\":\"@luk\"},{\"key\":\"channel\",\"value\":\"#lenses\"}]"`)
+	bite.CanBeSilent(cmd)
+	bite.Prepend(cmd, bite.FileBind(&channel))
+
+	return cmd
+}
+
+// NewUpdateAlertChannelCommand creates `alertchannels create` command
+func NewUpdateAlertChannelCommand() *cobra.Command {
+	var (
+		propertiesRaw string
+		channelID     string
+		channel       = api.AlertChannelPayload{}
+	)
+
+	cmd := &cobra.Command{
+		Use:              "update",
+		Short:            "Update an existing alert channel",
+		Example:          `alertchannels update --id="fa0e9b96-1048-4f4c-b776-4e96ca62f37d" --name="kafka-prd-health" --templateName="Slack" --connectionName="slack-connection" --properties="[{\"key\":\"username\",\"value\":\"@luk\"},{\"key\":\"channel\",\"value\":\"#lenses\"}]"`,
+		SilenceErrors:    true,
+		TraverseChildren: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			if err := bite.CheckRequiredFlags(cmd, bite.FlagPair{"id": channelID, "name": channel.Name, "connectionName": channel.ConnectionName, "templateName": channel.TemplateName, "properties": channel.Properties}); err != nil {
+				return err
+			}
+
+			if propertiesRaw != "" {
+				if err := bite.TryReadFile(propertiesRaw, &channel.Properties); err != nil {
+					// from flag as json.
+					if err = json.Unmarshal([]byte(propertiesRaw), &channel.Properties); err != nil {
+						return fmt.Errorf("Unable to unmarshal the properties: [%v]", err)
+					}
+				}
+			}
+
+			if err := config.Client.UpdateAlertChannel(channelID, channel.Name, channel.ConnectionName, channel.TemplateName, channel.Properties); err != nil {
```

Comment:
why you don't pass the whole object `channel` instead?

Reply:
Done!